internal/service/clouduserprojectassignment/resource_migration_test.go

```diff
@@ -0,0 +1,138 @@
+package clouduserprojectassignment_test
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/mig"
+)
+
+const (
+	resourceInvitationName            = "mongodbatlas_project_invitation.mig_test"
+	resourceProjectName               = "mongodbatlas_project.mig_test"
+	resourceUserProjectAssignmentName = "mongodbatlas_cloud_user_project_assignment.user_mig_test"
+)
+
+func TestMigCloudUserProjectAssignmentRS_basic(t *testing.T) {
+	mig.SkipIfVersionBelow(t, "2.0.0") // when resource 1st released
+	mig.CreateAndRunTest(t, basicTestCase(t))
+}
+
+func TestMigCloudUserProjectAssignmentRS_migrationJourney(t *testing.T) {
+	var (
+		orgID       = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		username    = acc.RandomEmail()
+		projectName = fmt.Sprintf("mig_user_project_%s", acc.RandomName())
+		roles       = []string{"GROUP_READ_ONLY", "GROUP_DATA_ACCESS_READ_ONLY"}
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acc.PreCheckBasic(t) },
+		CheckDestroy: checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: mig.ExternalProviders(),
+				Config:            originalConfigFirst(username, projectName, orgID, roles),
+			},
+			{
+				ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+				Config:                   userProjectAssignmentConfigSecond(username, projectName, orgID, roles),
+				Check:                    checksSecond(username, roles),
+			},
+			{
+				ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(resourceInvitationName, plancheck.ResourceActionDestroy),
+					},
+				},
+				Config: removeProjectInvitationConfigThird(username, projectName, orgID, roles),
+			},
+			mig.TestStepCheckEmptyPlan(removeProjectInvitationConfigThird(username, projectName, orgID, roles)),
+		},
+	})
+}
+
+func originalConfigFirst(username, projectName, orgID string, roles []string) string {
```

Comment:
```suggestion
func legacyProjectInvitationConfig(username, projectName, orgID string, roles []string) string {
```
Would make more explicit this is using the deprecated resource